src/java/org/apache/cassandra/cql3/Relation.java

```diff
@@ -17,227 +17,171 @@
-     * otherwise.
+     * Creates a relation for a single column (e.g. {@code columnA IN ?} ).
+     *
+     * @param identifier the column identifier to which the relation applies
+     * @param operator the relation operator
+     * @param rawTerms the terms to which the column values must be compared
+     * @return a relation for a single column.
      */
-    public final boolean isEQ()
-    {
-        return operator == Operator.EQ;
-    }
-
-    public final boolean isLIKE()
+    public static Relation singleColumn(ColumnIdentifier identifier, Operator operator, Terms.Raw rawTerms)
     {
-        return operator == Operator.LIKE_PREFIX
-               || operator == Operator.LIKE_SUFFIX
-               || operator == Operator.LIKE_CONTAINS
-               || operator == Operator.LIKE_MATCHES
-               || operator == Operator.LIKE;
+        return new Relation(ColumnsExpression.Raw.singleColumn(identifier), operator, rawTerms);
     }
 
     /**
-     * Checks if the operator of this relation is a <code>Slice</code> (GT, GTE, LTE, LT).
+     * Creates a relation for a map element (e.g. {@code columnA[?] = ?}).
      *
-     * @return <code>true</code> if the operator of this relation is a <code>Slice</code>, <code>false</code> otherwise.
+     * @param identifier the map column identifier
+     * @param rawKey the map element key
+     * @param operator the relation operator
+     * @param rawTerm the term to which the map element must be compared
+     * @return a relation for a map element.
      */
-    public final boolean isSlice()
+    public static Relation mapElement(ColumnIdentifier identifier, Term.Raw rawKey, Operator operator, Term.Raw rawTerm)
     {
-        return operator == Operator.GT
-               || operator == Operator.GTE
-               || operator == Operator.LTE
-               || operator == Operator.LT;
+        return new Relation(ColumnsExpression.Raw.mapElement(identifier, rawKey), operator, Terms.Raw.of(rawTerm));
     }
 
     /**
-     * Converts this <code>Relation</code> into a <code>Restriction</code>.
+     * Creates a relation for multiple columns (e.g. {@code (columnA, columnB) = (?, ?)}).
      *
-     * @param table the Column Family meta data
-     * @param boundNames the variables specification where to collect the bind variables
-     * @return the <code>Restriction</code> corresponding to this <code>Relation</code>
-     * @throws InvalidRequestException if this <code>Relation</code> is not valid
+     * @param identifiers the columns identifiers
+     * @param operator the relation operator
+     * @param rawTerm the term (tuple) to which the multiple columns must be compared
+     * @return a relation for multiple columns.
      */
-    public final Restriction toRestriction(TableMetadata table, VariableSpecifications boundNames)
+    public static Relation multiColumns(List<ColumnIdentifier> identifiers, Operator operator, Term.Raw rawTerm)
```

Comment:
```suggestion
    public static Relation multiColumn(List<ColumnIdentifier> identifiers, Operator operator, Term.Raw rawTerm)
```
...just to be consistent w/ `singleColumn` I guess